fplll/bkz.h

```diff
@@ -55,26 +78,107 @@ template <class ZT, class FT> class BKZAutoAbort
 /* The matrix must be LLL-reduced */
 template <class ZT, class FT> class BKZReduction
 {
+  /**
+   * @brief The class performing block reduction.
+   *
+   * This class implements BKZ, SD-BKZ and Slide Reduction. For this
+   * it relies on the GSO, LLL, and Enumeration modules. It assumes
+   * that the basis is LLL reduced.
+   *
+   * @param m
+   *    GSO object corresponding to the basis to be reduced
+   * @param lll_obj
+   *    LLL object associated to the same GSO object m
+   * @param param
+   *    parameter object (see bkz_param.h)
+   */
 public:
   BKZReduction(MatGSO<ZT, FT> &m, LLLReduction<ZT, FT> &lll_obj, const BKZParam &param);
   ~BKZReduction();
 
+  /**
+   * @brief Preprocesses a block
+   *
+   * Preprocess a block using LLL or stringer recursive preprocessing.
+   *
+   * @param kappa
+   *    start of the block
+   * @param block_size
+   *    size of the block
+   * @param param
+   *    parameter object
```

Comment:
coud be a more explicit. I always get confused, is that the param for the recursioned BKZ, or the current level of BKZ ?